searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp

```diff
@@ -267,16 +262,16 @@ struct MyFastAccessFeedView
         ISummaryAdapter::SP summaryAdapter(new MySummaryAdapter());
         Schema::SP schema(new Schema());
         _dmsc = make_shared<DocumentMetaStoreContext>(std::make_shared<BucketDBOwner>());
-        _lidReuseDelayer.reset(new documentmetastore::LidReuseDelayer(_writeService, _dmsc->get()));
+        _lidReuseDelayer = std::make_unique<documentmetastore::LidReuseDelayer>(_writeService, _dmsc->get());
         std::shared_ptr<const DocumentTypeRepo> repo = createRepo();
         StoreOnlyFeedView::Context storeOnlyCtx(summaryAdapter, schema, _dmsc, *_gidToLidChangeHandler, repo,
-                                                _writeService, *_lidReuseDelayer, _commitTimeTracker);
+                                                _writeService, *_lidReuseDelayer);
         StoreOnlyFeedView::PersistentParams params(1, 1, DocTypeName(DOC_TYPE), 0, SubDbType::NOTREADY);
         auto mgr = make_shared<AttributeManager>(BASE_DIR, "test.subdb", TuneFileAttributes(), _fileHeaderContext,
                                                  _writeService.attributeFieldWriter(), _writeService.shared(), _hwInfo);
-        IAttributeWriter::SP writer(new AttributeWriter(mgr));
+        auto writer = std::make_shared<AttributeWriter>(mgr);
```

Comment:
Use of auto here triggers a memory use after free error (detected by address sanitizer).

Reply:
How does that change scope ?

Reply:
A reference to a temporary std::shared_pointer<IattributeWriter> is passed to FastAccessFeedView::Context constructor and goes out of scope right afterwards, causing fastUpdateCtx to reference a dead stack variable.

Reply:
Fixed implementation of ...::Context to use shared_ptr by value instead.